pymeasure/instruments/advantest/advantestR624X.py

```diff
@@ -0,0 +1,2086 @@
+            - In sample hold mode, the AD transformation is just before the fall
+              of the pulse width.
+            - The sample hold mode cannot be set during DC measurement and DC sweep
+              measurement. When set to sample-and-hold mode, the integration time is 100 µs.
+              However, in 2-channel synchronous operation, if one channel is in pulse
+              generation and the other is in sample-and-hold mode, the DC measurement
+              side also operates in sample-and-hold mode.
+            - When performing pulse measurement and pulse sweep measurement, it
+              is necessary to satisfy the restrictions on the pulse width (Tw),
+              pulse period (Tp), and measure delay time (Td) of the WT command.
+              If the constraint is not satisfied, the integration time is unchanged.
+              To lengthen the integration time, first change the pulse width (Tw)
+              and pulse period (Tp). When shortening the pulse width and pulse
+              cycle, shorten the integration time first.
+
+        """,
+        validator=strict_discrete_set,
+        values=[0, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17],
+        get_process=lambda v: SampleHold(int(v)),
+    )
+
+    def set_sample_mode(self, mode, auto_sampling=True):
+        """ Sets synchronous, asynchronous, tracking operation
+        and search measurement between channels (``JM``).
+
+        :param mode: Sample Mode
+        :type mode: :class:`.SampleMode`
+        :param auto_sampling: Whether or not auto sampling is enabled, defaults to True
+        :type auto_sampling: bool, optional
+
+        """
+        mode = SampleMode(mode)
+        auto_sampling = map_values(auto_sampling, {True: 1, False: 2})
+
+        self.write(f'jm {mode.value},{auto_sampling},{{ch}}')
+
+    def set_timing_parameters(self, hold_time, measurement_delay, pulsed_width, pulsed_period):
+        """ Set the hold time, measuring time, pulse width and the pulse period (``WT``).
+
+        :param float hold_time: total amount of time for the complete pulse, until next pulse comes
+        :param float measurement_delay: time between measurements
+        :param float pulsed_width: Time specifying the pulse width
+        :param float pulsed_period: Time specifying the pulse period
+
+        .. note::
+
+            Pulse measurement has the following restrictions depending on the pulse period (Tp)
+            setting. (For pulse sweep measurements, there are no restrictions.)
+
+                - Tp < 2ms : Not measured.
+                - 2ms <= Tp < 10ms : Measure once every 5 ~ 20ms.
+                - 10ms <= Tp: Measured at each pulse generation.
+
+        """
+        self.write(f"wt {{ch}},{hold_time:.4e},{measurement_delay:.4e},{pulsed_width:.4e},"
+                   "{pulsed_period:.4e}")
+
+    def select_for_output(self):
+        """ This is a query command to select and for the channel to
```

Comment:
A typo with "and for"?
Maybe it is clearer with "to select a channel and"

Reply:
Agree, I'll change this